[DonnonsBridge] Add a new bridge

Binnette opened the pull request:
This bridge allows french users to scrap pages from https://donnons.org/

I test it since 2 month on my own server and it works nicely.

Before submiting my MR, i checked that my code and commits were following code styling rules.

Any comments or suggestions are very welcome since this is my first "bridge".

Binnette committed:
[DonnonsBridge] v0.1 - Duplicate code from existing bridge

Binnette committed:
[DonnonsBridge] v0.2 - Change class name

Binnette committed:
[DonnonsBridge] v0.3 - Fix php file name

Binnette committed:
[DonnonsBridge] v0.4 - Test bridge with hardcoded values

Binnette committed:
[DonnonsBridge] v1.0 - Fully working bridge

Binnette committed:
[DonnonsBridge] v1.1 - Add param 'pages to scan' with default value 5

Binnette committed:
[DonnonsBridge] v1.2 - Fix code styling

Binnette committed:
[DonnonsBridge] v1.3 - Fix query param url

Binnette committed:
[DonnonsBridge] v1.4 - Fix errors from phpcs

Binnette committed:
[DonnonsBridge] Merge branch 'master' into DonnonsBridge

Binnette committed:
[DonnonsBridge] Merge branch 'master' into DonnonsBridge

teromene commented on bridges/DonnonsBridge.php:
```diff
@@ -0,0 +1,111 @@
+
+	public function getURI() {
+		if(!is_null($this->getInput('q'))) {
+			return self::URI . $this->getInput('q');
```
You might wanna check whether the input starts with a "/". If not, the input does not validate. For this, you can use the "pattern" option of the PARAMETERS array.

Binnette replied:
Done in c21a304

teromene reviewed (changes requested):
Just a little nitpick.
Apart for that, the code is really nice and will be ready to merge. Sorry for the delay.

Binnette committed:
[DonnonsBridge] v1.5 - Fix error when uri already have query parameters

Binnette committed:
[DonnonsBridge] v1.6 - Add a pattern to check that url starts with /

Binnette commented:
Hi @teromene, thanks you for your reply. I added the pattern as asked (see commit c21a304).

Donnons.org recently changed their query parameters, so i just fixed what i think is the last error in my bridge with commit 836dc28.

I will test my "new version" a few day, and tell you when it's ok to merge my bridge into main project.

Have a nice day.

Binnette commented:
Hi @teromene, all my tests are OK. You can merge my bridge 👍

teromene commented:
Looks great to me ! Thank you for the PR and for the thorough testing !
